app/src/main/java/com/andrew/tasky/presentation/event_detail/EventDetailFragment.kt

```diff
@@ -0,0 +1,723 @@
+                viewModel.addPhoto(it)
+            }
+        }
+    )
+    private val args: EventDetailFragmentArgs by navArgs()
+    private val isAttendee = true
+    private val agendaItemType = AgendaItemType.EVENT
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding = FragmentEventDetailBinding.bind(view)
+        startTimeDateBinding = CvTimeDateSelectorBinding.bind(binding.startTimeAndDateLayout)
+        endTimeDateBinding = CvTimeDateSelectorBinding.bind(binding.endTimeAndDateLayout)
+        reminderLayoutBinding = CvReminderLayoutBinding.bind(binding.reminderLayout)
+        navController = Navigation.findNavController(view)
+        viewModel = ViewModelProvider(this)[EventDetailViewModel::class.java]
+
+        setupAgendaInitialViewModels()
+        setupHeader()
+        setupDoneButton()
+        setupTitleLayout()
+        setupDescriptionLayout()
+        setupPhotoLayout()
+        setupStartTimeLayout()
+        setupEndTimeLayout()
+        setupReminderTimesLayout()
+        setupAttendeeLayout()
+        setupDeleteJoinLeaveEventButton()
+
+        val attendeesList = listOf(
+            Attendee(name = "Samantha Jones", isAttending = true, attendeeType = AttendeeType.CREATOR, email = "random@gmail.com"),
+            Attendee(name = "Cappuccino Joe", isAttending = true, attendeeType = AttendeeType.ATTENDEE, email = "random1@gmail.com"),
+            Attendee(name = "Autumn Leaves", isAttending = true, attendeeType = AttendeeType.ATTENDEE, email = "random2@gmail.com"),
+            Attendee(name = "Andrew", isAttending = true, attendeeType = AttendeeType.ATTENDEE, email = "random3@gmail.com"),
+            Attendee(name = "Ramsay Beans", isAttending = false, attendeeType = AttendeeType.ATTENDEE, email = "random4@gmail.com"),
+            Attendee(name = "I Heart Lucy", isAttending = false, attendeeType = AttendeeType.ATTENDEE, email = "random5@gmail.com"),
+            Attendee(name = "I Have A Long name", isAttending = true, attendeeType = AttendeeType.ATTENDEE, email = "random6@gmail.com")
+        )
+
+        for (name in attendeesList){
+            viewModel.addAttendee(name)
+        }
+    }
+
+    private fun setupAgendaInitialViewModels() {
+        if (!viewModel.isInitiallySetup.value) {
+            args.agendaItem?.let {
+                viewModel.setTitle(it.title)
+                viewModel.setDescription(it.description)
+                viewModel.setStartDate(it.startDateAndTime.toLocalDate())
+                viewModel.setStartTime(it.startDateAndTime.toLocalTime())
+                viewModel.setIsDone(it.isDone)
+                viewModel.setSelectedReminderTime(it.reminderTime)
+                it.photos?.let { photoList -> viewModel.setupPhotos(photoList) }
+            }
+            viewModel.setEditMode(args.isInEditMode)
+            viewModel.setInitialSetupToTrue()
+        }
+    }
```

Comment:
Normally, this is stuff shouldn't need to initialize in the fragment since it's all ViewModel code. You can retrieve navigation arguments in a ViewModel like this:
```
class EventDetailViewModel(
    private val savedStateHandle: SavedStateHandle
): ViewModel() {
    
    init {
        savedStateHandle.get<AgendaItem>("agendaItem")?.let { item -> 
            // Update all the states here
        }
    }
}
```

Normally, the savedStateHandle instance should be passed automatically to your VM when you construct it using `by viewModels()` (not 100% on this anymore, if this doesn't work for you let me know)